### PR TITLE
New tests and improved error handling

### PR DESCRIPTION
**Pull request by G07cha**
I just wanted to setup few test for `dir` argument, but after I've done this I've noticed that one of modules just throws error if `giturl` variable incorrect(`null` for example) so I have added error handler as in sync call but for callback.

### CODE REVIEW

**Comment by jonschlinkert** [comment]
thanks! fwiw, this repo and remote-origin-url were originally created to be used with template helpers for a project I was doing a while back. the convention is (was) to usually not throw an error if a variable didn't resolve...

File: test.js [hunk 1]
```diff
@@ -19,3 +19,24 @@ describe('sync', function () {
     assert(repoName.sync() === 'git-repo-name');
   });
 });
+
+describe('dir', function() {
```
Comment (jonschlinkert) [inline test.js:23]: thanks for these!

File: index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ module.exports = function(dir, cb) {
 
   utils.origin(utils.cwd(dir), function (err, giturl) {
     if (err) return cb(err);
-
+    if(!giturl) return cb('cannot find ".git/config"');
```
Comment (jonschlinkert) [inline index.js]: maybe we should just make this a real error... e.g.

``` js
cb(new Error('cannot find ".git/config"'));
```

thoughts?

Comment (G07cha) [inline index.js]: I don't see big difference, but I'll do it.

Comment (jonschlinkert) [inline index.js]: thanks!

**Comment by G07cha** [comment]
Done

**Comment by G07cha** [comment]
@jonschlinkert is there another things to do before merging PR?

**Comment by jonschlinkert** [comment]
no sorry thanks for the reminder!

**Comment by jonschlinkert** [comment]
published and tagged as 0.4.1. thanks again!

**Comment by G07cha** [comment]
@jonschlinkert, you're welcome!

**Pull request merged**